lib/delighted/list_resource.rb

```diff
@@ -0,0 +1,41 @@
+module Delighted
+  class ListResource
+    def initialize(klass, path, opts, client)
+      @class = klass
+      @path = path
+      @opts = opts
+      @client = client
+      @iteration_count = 0
+    end
+
+    def auto_paging_each(opts = {})
+      auto_handle_rate_limits = opts.fetch(:auto_handle_rate_limits, true)
+      loop do
+        begin
+          # Get next (or first) page
+          if @iteration_count == 0
+            data = @client.request_get(@path, { params: @opts })
+          else
+            data = @client.request_get(@next_link, { full_url: true })
+          end
+        rescue Delighted::RateLimitedError => e
+          if auto_handle_rate_limits
+            sleep e.response.headers['Retry-After'].to_i
+            retry
+          else
+            raise
+          end
+        end
+
+        @iteration_count += 1
+        @next_link = data[:response].next_link
+
+        data[:json].map do |attributes|
+          yield Utils.full_const_get(@class).new(attributes)
```

Comment:
Maybe move the `Utils.full_const_get(@class)` into the `initialize` method? In addition to reducing duplicate work, the code would fail faster if a bad `klass` value were passed.

Actually, could we avoid the need for `Utils.full_const_get` entirely by passing the actual class object rather than the name: `ListResource.new(self, path, Utils.serialize_values(opts), client)`?

Reply:
Your second point is way better, using that 👍